samples/n5010-ctl/n5010-ctl.c

```diff
@@ -0,0 +1,504 @@
+}
+
+static fpga_result fpga_base(struct n5010 *n5010)
+{
+	fpga_result res;
+
+	while (1) {
+		uint64_t header;
+		struct {
+			union {
+				uuid_t u128;
+				uint64_t u64[2];
+			};
+		} uuid;
+		int i;
+
+
+		res = fpgaReadMMIO64(n5010->handle, 0, n5010->base, &header);
+		if (res != FPGA_OK) {
+			fprintf(stderr, "failed to read dfh: %s\n",
+				fpgaErrStr(res));
+			break;
+		}
+
+		res = fpgaReadMMIO64(n5010->handle, 0, n5010->base + 8,
+				     &uuid.u64[1]);
+		if (res != FPGA_OK) {
+			fprintf(stderr, "failed to read uuid high: %s\n",
+				fpgaErrStr(res));
+			break;
+		}
+
+		res = fpgaReadMMIO64(n5010->handle, 0, n5010->base + 16,
+				     &uuid.u64[0]);
+		if (res != FPGA_OK) {
+			fprintf(stderr, "failed to read uuid low: %s\n",
+				fpgaErrStr(res));
+			break;
+		}
+
+		for (i = 0; i < 2; i++)
+			uuid.u64[i] = be64toh(uuid.u64[i]);
+
+		if (DFH_TYPE(header) == DFH_TYPE_AFU
+		    && uuid_compare(uuid.u128, n5010->guid) == 0) {
+		}
+		break;
+
+		if (DFH_EOL(header)) {
+			fprintf(stderr, "no matching dfh found: 0x%016jx\n",
+				header);
+			res = FPGA_NOT_FOUND;
+			break;
+		}
+
+		if (DFH_NEXT(header) == 0 || DFH_NEXT(header) == 0xffff) {
+			fprintf(stderr, "next dfh not found: 0x%016jx\n",
+				header);
+			res = FPGA_NOT_FOUND;
+			break;
+		}
+
+		n5010->base += DFH_NEXT(header);
+	}
+
+	return res;
+}
+
+static void fpga_dump(struct n5010 *n5010, uint64_t offset, size_t count)
+{
+	uint64_t base = n5010->base + offset;
+
+	if (!n5010->debug)
+		return;
+
+	for (size_t i = 0; i < count; i++) {
+		uint64_t reg = base + i * 8;
+		uint64_t val;
+
+		fpgaReadMMIO64(n5010->handle, 0, reg, &val);
+		printf("reg: 0x%04jx, val: 0x%016jx\n", reg, val);
+	}
+}
+
+
+static fpga_result fpga_set_switch(struct n5010 *n5010)
+{
+	fpga_result res;
+
+	uint64_t offset = CVL0_QSFP01_SWITCH;
+	uint64_t port = n5010->port;
+	if (n5010->port > 1) {
+		port -= 2;
+		offset = CVL1_QSFP23_SWITCH;
+	}
+
+
+	res = fpgaWriteMMIO64(n5010->handle, 0, n5010->base + offset, port);
+	if (res != FPGA_OK) {
+		fprintf(stderr, "failed to set switch: %s\n", fpgaErrStr(res));
+		goto error;
+	}
+
+error:
+	return res;
+}
+
+static fpga_result fpga_read_switch_port(struct n5010 *n5010, uint32_t cvl)
+{
+	uint64_t port = 0;
+	uint64_t offset = CVL0_QSFP01_SWITCH;
+
+	if (cvl == 1)
+		offset = CVL1_QSFP23_SWITCH;
+
+	fpgaReadMMIO64(n5010->handle, 0, n5010->base + offset, &port);
+	printf("CVL: %u connected to front port: %lu\n", cvl, port + (cvl * 2));
+	return 0;
+}
+
+static fpga_result fpga_read_switch(struct n5010 *n5010)
+{
+	fpga_read_switch_port(n5010, 0);
+	fpga_read_switch_port(n5010, 1);
+	return 0;
+}
+
+static fpga_result fpga_setread_switch(struct n5010 *n5010)
+{
+	fpga_result res;
+
+	if (n5010->port == INVALID_PORT) {
+		fprintf(stderr, "no port given\n");
+		return EXIT_FAILURE;
+	}
+
+	res = fpga_set_switch(n5010);
+	fpga_read_switch(n5010);
+	return res;
+}
+
+static fpga_result fpga_run(struct n5010 *n5010)
+{
+	if (n5010->debug) {
+	};
```

Comment:
Please remove debug/dead code.

Reply:
Exchanged with __attribute__((unused)). This was a dummy use of unused parameter because function pointer have type.